### PR TITLE
Remove use of update() in funccount

### PR DESCRIPTION
**Pull request by drzaeus77**
This is an alternative fix for #415, funccount being called in a recursive context.

### CODE REVIEW

File: src/python/bcc/table.py [hunk 1]
```diff
@@ -186,6 +186,10 @@ def clear(self):
         for k in self.keys():
             self.__delitem__(k)
 
+    def zero(self):
+        for k in self.keys():
+            self[k] = self.Leaf()
```
Comment (4ast) [inline src/python/bcc/table.py]: I don't think this part actually clear it in a hash map.
Isn't zero-ing it in python memory only?

Comment (drzaeus77) [inline src/python/bcc/table.py]: No, everything implemented in this file is write-through.

Comment (drzaeus77) [inline src/python/bcc/table.py]: To be specific, this will invoke __setitem__(key, value)

Comment (4ast) [inline src/python/bcc/table.py]: yeah found it. so it will call update_elem() which will do kmalloc in user context which is ok.

Comment (4ast) [inline src/python/bcc/table.py]: hmm then delete should be ok as well.

File: tools/funccount.py [hunk 2]
```diff
@@ -81,6 +85,16 @@ def signal_ignore(signal, frame):
...
 b = BPF(text=bpf_text)
+counts = b.get_table("counts")
+
+# pre-insert the function addresses into the counts table
+fns = b._get_kprobe_functions(pattern)
+for fn in fns:
+    addr = b.ksymname(fn)
+    if addr == -1:
+        raise Exception("Unknown symbol name %s" % fn)
+    counts[counts.Key(addr)] = counts.Leaf()
```
Comment (4ast) [inline tools/funccount.py]: without checking for collisions this approach is no better than converting array.
can you add collision check here?

Comment (drzaeus77) [inline tools/funccount.py]: Collision at which level? Two ksyms can have the same address (SyS_write and sys_write), but the address itself will have a unique entry in the bpf hlist underlying the map, correct? Are you suggesting to use BPF_NOEXIST? I don't think that would matter here.

Comment (4ast) [inline tools/funccount.py]: got it. yeah should work.

**Comment by goldshtn** [comment]
How would you propose to deal with this issue in argdist? Unlike here, the keys are actually arbitrary structures and aren't known in advance...

**Pull request merged**